sympy/codegen/lfort.py

```diff
@@ -0,0 +1,85 @@
+from sympy.printing.codeprinter import CodePrinter
+
+import lfortran.asr.asr as asr
+import lfortran.asr.builder as builder
+
+
+class ASRConverter(CodePrinter):
+
+    def __init__(self):
+
+        # The translation unit that will be modified during the conversion
+        # process. Since this object is stateful, the intention is that each
+        # instance of this class is used only once.
+        self.translation_unit = builder.make_translation_unit()
+        self.type_integer = builder.make_type_integer()
+
+        # Variables that will be used in the body of a function declaration.
+        self.variables = []
+
+    def _print(self, expr, **kwargs):
+        convertmethod = '_print_' + type(expr).__name__
+        if hasattr(self, convertmethod):
+            return getattr(self, convertmethod)(expr, **kwargs)
+        else:
```

Comment:
This `else` isn't needed.

Reply:
Is it actually necessary to define this method at all? It should be handled in the superclass.

Reply:
It works with the superclass method, but opens up the possibility of the printer returning something other than an AST. I've gotten rid of it for now, since it seems this can be adjusted in the superclass settings.